### PR TITLE
Update libpod to support labels

### PR DESCRIPTION
**Pull request by mheon**
Also clean up properly if NewRuntime fails.

I think this may be required for the create frontend PR.

### CODE REVIEW

File: libpod/options.go [hunk 1]
```diff
@@ -320,12 +320,18 @@ func (r *Runtime) WithPod(pod *Pod) CtrCreateOption {
...
 // WithLabels adds labels to the container
 func WithLabels(labels map[string]string) CtrCreateOption {
-	return ctrNotImplemented
-}
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
 
-// WithAnnotations adds annotations to the container
```
Comment (rhatdan) [inline libpod/options.go]: Are we not implementing WithAnnotations?

**Comment by rhatdan** [comment]
LGTM other then quiestion on annotations.

File: libpod/options.go (hunk 1 above)
Comment (mheon) [reply libpod/options.go]: Annotations are stored directly in the spec. We already store the spec. No reason to store additional annotations elsewhere.

**Comment by rhatdan** [comment]
Passes tests.  Merging.

**Pull request merged**